### ISSUE TITLE
New transparency blending is excessively saturated on light background + light wallpaper

### ISSUE THREAD

**Issue by Thesola10** [kovidgoyal/kitty#6250]
**Describe the bug**
Lighter colors are distorted on semi-transparent kitty windows

**To Reproduce**
Steps to reproduce the behavior (GNOME, both X11 and Wayland):
1. `background #C3D4EE`
2. `background_opacity 0.7`
3. Wallpaper contains `#94BCE5`
4. colors go full klebold

**Screenshots**

**Buggy screenshot**
![kitty bad](https://user-images.githubusercontent.com/7287268/236614617-6ee36f00-2bf7-42db-82cc-837bf8d38ef3.png)

**Correct screenshot (0.27.x)**
![kitty good](https://user-images.githubusercontent.com/7287268/236614655-26821c6d-b8a9-497b-a83b-39f8189248f0.png)

**Environment details**
```
kitty 0.28.1 created by Kovid Goyal
Linux SolaXBook 6.3.1-arch1-1 #1 SMP PREEMPT_DYNAMIC Mon, 01 May 2023 17:42:39 +0000 x86_64
Arch Linux 6.3.1-arch1-1 (/dev/tty)

DISTRIB_ID="Arch"
DISTRIB_RELEASE="rolling"
DISTRIB_DESCRIPTION="Arch Linux Atomic"
Running under: Wayland
Frozen: False
Paths:
  kitty: /usr/bin/kitty
  base dir: /usr/lib/kitty
  extensions dir: /usr/lib/kitty/kitty
  system shell: /bin/zsh
Loaded config files:
  /home/karim/.config/kitty/kitty.conf

Config options different from defaults:
allow_remote_control       yes
background_opacity         0.7
bold_font                  FiraCode-Bold
bold_italic_font           Comic Code Ligatures Bold Italic
click_interval             0.5
cursor_blink_interval      0.5
dynamic_background_opacity True
font_family                FiraCode-Regular
font_features:
{'FiraCode-Bold': ('+ss07',), 'FiraCode-Regular': ('+ss07',)}
font_size                  9.0
initial_window_height      (365, 'px')
initial_window_width       (615, 'px')
italic_font                Comic Code Ligatures Italic
listen_on                  unix:@kitty
remember_window_size       False
resize_debounce_time       0.2
resize_draw_strategy       3
scrollback_pager           ['less', '+G', '-R']
select_by_word_characters  :@-./_~?&=%+#
symbol_map:
	U+3000 - U+30ff → Noto Sans CJK JP
	U+4e00 - U+9fff → Noto Sans CJK JP
	U+e000 - U+faff → Hack Nerd Font
	U+ff00 - U+ffef → Noto Sans CJK JP
touch_scroll_multiplier    5.0
wayland_titlebar_color     1
window_border_width        (0, 'pt')
Added mouse actions:
	ctrl+shift+left click grabbed →  mouse_handle_click link
	ctrl+shift+left click ungrabbed →  mouse_handle_click link
Changed mouse actions:
	left click ungrabbed →  mouse_handle_click selection prompt
Changed shortcuts:
	kitty_mod+- →  decrease_font_size
	kitty_mod+= →  increase_font_size
	kitty_mod+backspace →  restore_font_size
Colors:
	background                 #c3d4ee   
	color0                     #073642   
	color1                     #ac122f   
	color10                    #586e75   
	color11                    #657b83   
	color12                    #839496   
	color13                    #6c71c4   
	color14                    #93a1a1   
	color15                    #808080   
	color2                     #356900   
	color3                     #b58900   
	color4                     #265ba2   
	color5                     #a31652   
	color6                     #1a7158   
	color7                     #606060   
	color8                     #002b36   
	color9                     #bb3b16   
	cursor                     #ffffff   
	foreground                 #000000   
	selection_background       #c0d0e6   
	selection_foreground       #303030   

Important environment variables seen by the kitty process:
	PATH                                /home/karim/.nix-profile/bin:/nix/var/nix/profiles/default/bin:/home/karim/.local/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/home/karim/.local/share/flatpak/exports/bin:/var/lib/flatpak/exports/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/lib/rustup/bin
	LANG                                fr_FR.UTF-8
	SHELL                               /bin/zsh
	GLFW_IM_MODULE                      ibus
	DISPLAY                             :0
	WAYLAND_DISPLAY                     wayland-0
	USER                                karim
	XDG_DATA_DIRS                       /home/karim/.nix-profile/share:/home/karim/.local/share:/home/karim/.local/share/flatpak/exports/share:/run/media/karim/Mothership/Flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share/:/usr/share/
	XDG_RUNTIME_DIR                     /run/user/60403
	XDG_CURRENT_DESKTOP                 GNOME
	XDG_MENU_PREFIX                     gnome-
	XDG_SESSION_CLASS                   user
	XDG_SESSION_DESKTOP                 gnome
	XDG_SESSION_TYPE                    wayland
```

**Additional context**
I'm running Intel UHD graphics with Mesa 23.0.3

see steps for minimal config. Also, note how the blended color gets even brighter than the titlebar color in the buggy screenshot

**Comment by Thesola10** [comment]
right, and translucent screenshots like those taken by GNOME, where the wallpaper isn't included...

well they look RED.

![Capture d’écran du 2023-05-06 11-18-11](https://user-images.githubusercontent.com/7287268/236615312-4a33fdfb-edbc-4b00-b520-8c07b531b81d.png)

**Comment by kovidgoyal** [comment]
dup of #6209

**Comment by Thesola10** [comment]
> dup of #6209
> 

I don't think so, this issue is about the issues caused by **the fix to** #6209

**Comment by kovidgoyal** [comment]
Then post there so the person that made those fixes is notified and can
follow up

**Comment by Thesola10** [comment]
given that these changes have made it to release I think this should be a project issue.
I can ping the author of the PR but as long as the _issue_ affects _upstream_ kitty then this ticket should remain open on _upstream_ kitty.

**Comment by kovidgoyal** [comment]
If you want some help/changes post in either the PR or the issue. That will move this forward, otherwise nothing will happen.

**Comment by Thesola10** [comment]
that's still no reason to close this as duplicate.

**Comment by Thesola10** [comment]
@kovidgoyal please reopen.

**Comment by kovidgoyal** [comment]
The red screenshots I cant replicate, that is likely some issue with whatever screenshot tool you are using. As for the distortion, the only distortion I can see is the background color is a bit lighter in 0.28.0. The fringing on the text is fixed by #6218. The lighter background is easily fixed by adjusting the background_opacity or the background color, whichever gives you a final appearance you like.

It's debatable whether we should apply a back-correction to background_opacity so that its effect on the final color is closer to pre 0.28 or is at least linear. Note that #6218 is not in 0.28.1 you will need to wait for 0.28.2 or use a nightly build. If there is something else I missed feel free to update.

Window screenshot under GNOME wayland attached with your settings above though and the default GNOME dark blue wallpaper.
![Screenshot from 2023-05-24 22-15-48](https://github.com/kovidgoyal/kitty/assets/1308621/954bf8c2-db4b-450d-842a-6a8cc5c0b971)
